pkg/lib/cockpit-components-inline-notification.tsx

```diff
@@ -26,17 +26,30 @@ import './cockpit-components-inline-notification.css';
 
 const _ = cockpit.gettext;
 
-function mouseClick(fun) {
-    return function (event) {
+function mouseClick(fun: (ev: React.MouseEvent) => void): (ev: React.MouseEvent) => void {
+    return function (event: React.MouseEvent) {
```

Comment:
These 2 added lines are not executed by any test.

Reply:
I just don't like this function.  This whole setup with mouseClick() and toggleDetail() and `this.toggleDetail = this.toggleDetail.bind(this)` is way too complicated.  Now is a good opportunity to clean this up.

My instinct is always towards inlining handlers with typescript to benefit from the inferred types for the parameters.  I tried a bit at that and was able to remove all of the complicated tangle.  I don't know if this works, but I managed to get to something like:

```tsx
        const detailButton = (detail &&
            <Button variant="link" isInline className='alert-link more-button'
                onClick={event => {
                    if (event.button !== 0)
                        return;
                    event.preventDefault();
                    this.setState({ isDetail: !this.state.isDetail });
                }}
            >
                {this.state.isDetail ? _("show less") : _("show more")}
            </Button>
        );
```

...at this point, the `detailButton` itself could probably be inlined into the main component, but that's a matter of taste...